technical-reports/color/token-naming.md

```diff
@@ -0,0 +1,381 @@
+
+<aside class="example">
+
+```scss
+$color-gray-22: hsl(0, 0%, 22%);
+$color-gray-49: hsl(0, 0%, 49%);
+$color-gray-73: hsl(0, 0%, 73%);
+$color-gray-99: hsl(0, 0%, 99%);
+```
+
+</aside>
+
+**Computer generated scales**
+
+Token names may also be generated by tools, where the user specifies the base name, and the tool appends scale numbers based on changes to the underlying value.
+
+<aside class="example">
+
+```scss
+// User specified name
+$color-green
+
+// Tool generated names for 6 steps of opacity
+$color-green-10: rgba(32,178,170, 0.10);
+$color-green-20: rgba(32,178,170, 0.20);
+$color-green-30: rgba(32,178,170, 0.30);
+$color-green-40: rgba(32,178,170, 0.40);
+$color-green-50: rgba(32,178,170, 0.50);
+$color-green-60: rgba(32,178,170, 0.60);
+```
+
+</aside>
+
+| Pros                                                                                                                                                                 | Cons                                                      |
+| -------------------------------------------------------------------------------------------------------------------------------------------------------------------- | --------------------------------------------------------- |
+| Easy mapping between different tokens for color contrast. For example, all 100-400 tokens pair with 500-900 tokens in order to create accessible color combinations. | Less memorable and less obvious difference between tokens |
+| If using bounded scales, the token name can help indicate something about the underlying value                                                                       |                                                           |
+
+### Alias Tokens
+
+For **Alias Tokens**, we recommend grouping tokens with similar intentions by prioritizing the category + property within the name. For example, all background color Alias tokens would likely start with `$color-background-XXX`.
+
+We recommend avoiding abbreviations. For example, use “background” instead of “bg”, to help with clarity.
+
+Here’s how alias tokens may be represented:
+
+<aside class="example">
+
+```scss
+$color-background-error: $color-red-600;
+$color-background-success: $color-green-400;
+$color-text-base: $color-palette-black; // Token with variant
+$color-text-error-hover: $color-red-700; // Token with variant and state
+```
+
+</aside>
+
+Color alias tokens could also be grouped by concept, like so:
+
+<aside class="example">
+
+```scss
+// These can be used for background, border, or text colors
+$color-feedback-error: $color-red-600;
+$color-feedback-success: $color-green-400;
+```
+
+</aside>
+
+### Component Tokens
+
+Component specific names should start with the component that they support, and be located close to the component code. They commonly refer to alias tokens under the hood, and can be helpful when trying to use consistent styles across components while still maintaining separation of concerns.
```

Comment:
```suggestion
Component-specific names should start with the component they support and be located close to the component code. They commonly refer to alias tokens and can be helpful when trying to use consistent styles across components while still maintaining separation of concerns.
```

I removed "under the hood", it's an idiom that international readers may not know the meaning of.